src/sage/schemes/elliptic_curves/kummer_line.py

```diff
@@ -0,0 +1,1765 @@
+        self._X, self._Z = coords
+
+        if not self.is_x_coord():
+            raise ValueError("not a valid x-coordinate")
+
+    def __repr__(self):
+        r"""
+        String representation of a Kummer point.
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(101), [2, 3])
+            sage: P = E(95, 49)
+            sage: K = KummerLine(E)
+            sage: xP = K(P); xP.__repr__()
+        """
+        return f"({self._X} : {self._Z})"
+
+    def __bool__(self):
+        r"""
+        Boolean value for a Kummer line point.
+
+        OUTPUT: ``False`` if this is the point at infinity, otherwise ``True``
+
+        .. SEEALSO::
+
+            meth:`is_zero`
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(101), [2, 3])
+            sage: P = E(95, 49)
+            sage: K = KummerLine(E)
+            sage: xO = K(None); bool(xO)
+            False
+            sage: xP = K(P); bool(xP)
+            False
+        """
+
+        return bool(self._Z)
+
+    def __eq__(self, other):
+        r"""
+        Equality of two Kummer points as `\PP^1` elements.
+
+        EXAMPLES::
+
+            sage: E1 = EllipticCurve(GF(101), [2, 3])
+            sage: K1 = KummerLine(E1)
+            sage: P = E1(95, 49)
+
+        Two points can have different coordinates and still be the same projectively::
+            sage: xP = K1(P1)
+            sage: xQ = K1([29, 12])
+            sage: xP == xQ  # 95*12 % 101 = 29
+            True
+
+        They must represent the same `x`-coordinate::
+            sage: Q = E1(50, 41)
+            sage: xP, xQ = K1(P), K1(Q)
+            sage: xP == xQ
+            False
+
+        Base Kummer line must be the same even if the `x`-coordinates are equal::
+            sage: E2 = EllipticCurve(GF(101), [3, 7])
+            sage: Q = E2(95, 50)
+            sage: K2 = KummerLine(E2)
+            sage: xP, xQ = K1(P), K2(Q)
+            sage: xP == xQ
+            False
+            sage: xP.x() == xQ.x()
+            True
+            sage: xP.parent() == xQ.parent()
+            False
+        """
+        if not isinstance(other, KummerLinePoint):
+            raise ValueError("can only compare equality between two Kummer points")
+
+        if self.parent() != other.parent():
+            return False
+
+        XP, ZP = self.XZ()
+        XQ, ZQ = other.XZ()
+        return XP * ZQ == XQ * ZP
+
+    def is_zero(self):
+        r"""
+        A Kummer Point is zero if it corresponds to the point at infinity
+        on the parent curve.
+
+        OUTPUT: ``False`` if this is the point at infinity, otherwise ``True``
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(101), [2, 3])
+            sage: P = E(95, 49)
+            sage: K = KummerLine(E)
+            sage: xO = K(None); xO.is_zero()
+            False
+            sage: xP = K(P); xP.is_zero()
+            False
+        """
+
+        return not bool(self)
+
+    def is_x_coord(self):
```

Comment:
Is my understanding correct that this will always return `True` on a valid Kummer Point? If so, this should either be a hidden method (`_is_x_coord`), or there should be a method/constructor that possibly returns an invalid point. For example, `EllipticCurve` has `.point(..., check=False)`.